src/core/qgspointlocator.cpp

```diff
@@ -792,6 +842,30 @@ void QgsPointLocator::onFeatureAdded( QgsFeatureId fid )
     if ( !f.hasGeometry() )
       return;
 
+    std::unique_ptr< QgsFeatureRenderer > renderer( mLayer->renderer() ? mLayer->renderer()->clone() : nullptr );
```

Comment:
It would be better to be cloned inside "if ( mContext )" block, so we do not copy it when snapping also to invisible features.